integration/airflow/openlineage/lineage_backend/__init__.py

```diff
@@ -43,17 +43,25 @@ def send_lineage(self, operator=None, inlets=None, outlets=None, context=None):
         dag = context["dag"]
         dagrun = context["dag_run"]
         task_instance = context["task_instance"]
-        dag_run_id = self.adapter.build_dag_run_id(dag.dag_id, dagrun.run_id)
+        dag_run_id = self.adapter.build_dag_run_id(
+            dag_id=dag.dag_id,
+            run_id=dagrun.run_id,
+            execution_date=dagrun.execution_date,
+        )
 
-        run_id = str(uuid.uuid4())
+        run_id = str(generate_new_uuid())
         job_name = get_job_name(operator)
 
         task_metadata = self.extractor_manager.extract_metadata(
             dagrun=dagrun, task=operator, complete=True, task_instance=task_instance
         )
 
         task_uuid = OpenLineageAdapter.build_task_instance_run_id(
-            dag.dag_id, operator.task_id, task_instance.execution_date, task_instance.try_number
+            dag_id=dag.dag_id,
+            task_id=operator.task_id,
+            run_id=task_instance.dag_run.run_id,
```

Comment:
For some reason, this does not exist in Airflow 2.1:

```
[2024-05-14 09:09:16,524] {taskinstance.py:1463} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1165, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 1286, in _prepare_and_execute_task_with_callbacks
    task_copy.post_execute(context=context, result=result)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/lineage/__init__.py", line 124, in wrapper
    _backend.send_lineage(operator=self, inlets=self.inlets, outlets=self.outlets, context=context)
  File "/app/openlineage/integration/airflow/openlineage/lineage_backend/__init__.py", line 115, in send_lineage
    return cls.backend.send_lineage(*args, **kwargs)
  File "/app/openlineage/integration/airflow/openlineage/lineage_backend/__init__.py", line 62, in send_lineage
    run_id=task_instance.dag_run.run_id,
AttributeError: 'NoneType' object has no attribute 'run_id'
```

However, instead of fixing this, we might just drop Airflow 2.1 support all along. It's not supported by any cloud provider - Astronomer, MWAA, GCC - not to mention OSS Airflow project.

Reply:
It looks like there was [get_dag_run](https://airflow.apache.org/docs/apache-airflow/2.1.4/_api/airflow/models/taskinstance/index.html#airflow.models.taskinstance.TaskInstance.get_dagrun) method, which then was replaced with a property/relationship in 2.2+.

Reply:
I can remove using run_id in UUID generation, execution_date already provides uniquene value

Reply:
Removed, use `execution_date` as a source for randomness for UUID least significant bytes